src/components/OverviewMap.js

```diff
@@ -147,52 +93,77 @@ class OverviewMap extends React.Component {
-              iconRetinaUrl: leafletMarkerRetinaIconUrl,
-              iconSize: [25, 41],
-              iconAnchor: [13, 41]
-            })}
-            {...this.getAdditionalParams(hearing)}
-          >{this.getPopupContent(hearing, geojson)}
-          </Marker>
-        );
-      }
-      case "LineString": {
-        const latLngs = geojson.coordinates.map(([lng, lat]) => new LatLng(lat, lng));
-        return (<Polyline key={Math.random()} positions={latLngs}>{this.getPopupContent(hearing, geojson)}</Polyline>);
-      }
-      case "Feature": {
-        /**
-         * Recursively get the Map element
-         */
-        return (this.getMapElement(geojson.geometry, hearing));
+    const {mapElementLimit} = this.props;
+    const {id} = hearing;
+    if (geojson) {
+      switch (geojson.type) {
+        case "Polygon": {
+          // XXX: This only supports the _first_ ring of coordinates in a Polygon
+          const latLngs = geojson.coordinates[0].map(([lng, lat]) => new LatLng(lat, lng));
+          return (
+            <Polygon
+              key={id + "" + Math.random()}
+              positions={latLngs}
+            >
+              {this.getPopupContent(hearing, geojson)}
+            </Polygon>);
+        }
+        case "Point": {
+          const latLngs = new LatLng(geojson.coordinates[1], geojson.coordinates[0]);
+          return (
+            <Marker
+              position={latLngs}
+              key={id + "" + Math.random()}
+              icon={new Leaflet.Icon({
+                iconUrl: leafletMarkerIconUrl,
+                shadowUrl: leafletMarkerShadowUrl,
+                iconRetinaUrl: leafletMarkerRetinaIconUrl,
+                iconSize: [25, 41],
+                iconAnchor: [13, 41]
+              })}
+              {...this.getAdditionalParams(hearing)}
+            >{this.getPopupContent(hearing, geojson)}
+            </Marker>
+          );
+        }
+        case "LineString": {
+          const latLngs = geojson.coordinates.map(([lng, lat]) => new LatLng(lat, lng));
+          return (
+            <Polyline key={id + "" + Math.random()} positions={latLngs}>
+              {this.getPopupContent(hearing, geojson)}
+            </Polyline>
+          );
+        }
+        case "FeatureCollection": {
+          // if mapElementLimit is true & more than 0, display up to that amount of elements
+          const mapElementArray = [];
+          if (mapElementLimit && mapElementLimit > 0) {
+            geojson.features.slice(0, mapElementLimit).forEach((feature) => {
+              mapElementArray.push(this.getMapElement(feature.geometry, hearing));
+            });
+          } else {
+            // if mapElementLimit is false -> display all elements found in FeatureCollection
+            geojson.features.forEach((feature) => {
+              mapElementArray.push(this.getMapElement(feature.geometry, hearing));
+            });
+            return mapElementArray; 
+          }
+        }
```

Comment:
`return mapElementArray` is currently inside the `else` part, move `return mapElementArray` to happen after the if / else sections (between curly brackets on lines 150 / 151).

Reply:
It does detect it but doesn't highlight it well with my current configurations, have to change it 👍